.dev/systemimage/climate_machine_image.jl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env julia
+#
+# Called with no arguments will create the system image
+#     ClimateMachine.so
+# in the `@__DIR__` directory.
+#
+# Called with a single argument the system image will be placed in the path
+# specified by the argument (relative to the callers path)
+sysimage_path =
+    isempty(ARGS) ? joinpath(@__DIR__, "ClimateMachine.so") : ARGS[1]
+
+using Pkg
+Pkg.add("PackageCompiler")
+using PackageCompiler
+Pkg.activate(joinpath(@__DIR__, "../../"))
+Pkg.instantiate()
+pkgs = Pkg.installed()
```

Comment:
Should this be constrained to just the whitelisted CliMA project dependencies? (ex. a user installs additional packages into their project environment could lead to divergent system images maybe causing confusion / debugging issues)

Reply:
The installed only includes packages from the project.

As seen below after the `Pkg.activate` both PackageCompiler and Revise are not listed by `Pkg.installed`

```
❯ julia -e "using Pkg; display(Pkg.installed()); println(); println();Pkg.activate(@__DIR__); println(); display(Pkg.installed());"
Dict{String,Union{Nothing, VersionNumber}} with 2 entries:
  "PackageCompiler" => v"1.1.1"
  "Revise" => v"2.6.4"

Activating environment at `~/codes/CLIMA/Project.toml`

Dict{String,Union{Nothing, VersionNumber}} with 34 entries:
  "Pkg" => nothing
  "Statistics" => nothing
  "ForwardDiff" => v"0.10.10"
  "Coverage" => v"1.0.0"
  "Distributions" => v"0.23.2"
  "MPI" => v"0.13.0"
  "CLIMAParameters" => v"0.1.3"
  "Random" => nothing
  "GaussQuadrature" => v"0.5.4"
  "CUDAdrv" => v"6.0.0"
  "OrderedCollections" => v"1.1.0"
  "Literate" => v"2.3.1"
  "DelimitedFiles" => nothing
  "LinearAlgebra" => nothing
  "DoubleFloats" => v"1.1.8"
  "GPUifyLoops" => v"0.2.9"
  "Combinatorics" => v"1.0.0"
  "CUDAapi" => v"3.1.0"
  "LazyArrays" => v"0.16.5"
  "CuArrays" => v"1.7.3"
  "DocStringExtensions" => v"0.8.1"
  "WriteVTK" => v"1.4.1"
  "CUDAnative" => v"2.10.2"
  "FileIO" => v"1.2.4"
  "StaticArrays" => v"0.12.1"
  "JLD2" => v"0.1.13"
  "NCDatasets" => v"0.10.2"
  "Dates" => nothing
  "SpecialFunctions" => v"0.10.0"
  "Logging" => nothing
  "Printf" => nothing
  "KernelAbstractions" => v"0.2.3"
  "RootSolvers" => v"0.2.0"
  "ArgParse" => v"1.1.0"
```